js/src/components/paid-ads/ads-campaign.js

```diff
@@ -55,6 +58,21 @@ export default function AdsCampaign( {
 				'google-listings-and-ads'
 		  );
 
+	const {
+		country,
+		dailyBudget,
+		multipleRecommendations,
+		recommendations,
+		loading,
+	} = useBudgetRecommendationData( formContext.values.countryCodes );
+
+	useEffect( () => {
+		if ( ! loading ) {
+			const { amount } = clientSession.getCampaign() || {};
+			formContext.setValue( 'amount', amount || dailyBudget );
+		}
+	}, [ loading ] );
```

Comment:
This is why the `eslint-disable` rule was added above. Looks like this is meant to only run once to set the initial value after we've finished loading the `dailyBudget` value. However, this causes two problems. 1) I think only running this when `loading` changes runs the risk of a race condition where this runs before the `dailyBudget` value is updated. 2) If you add those dependencies back, each time you update the formContext value, it will cause the `formContext` value to change and cause a loop. To avoid this, we can update this to also only set the form value if it hasn't already been set. Here's an example that seems to be working for me locally. It assumes that the starting amount will always be 0 or falsey, so once it's set to a positive value, we can assume we've already set the value and not rerun the setter.

```js
	useEffect( () => {
		const setFormValue = ( value ) => {
			formContext.setValue( 'amount', value );
		};

		const { amount } = formContext.values;

		if ( ! loading && ! amount ) {
			const { savedAmount } = clientSession.getCampaign() || {};
			setFormValue( savedAmount || dailyBudget );
		}
	}, [ dailyBudget, formContext, loading ] );
```